scripts/main.js

```diff
@@ -49,11 +46,14 @@ function update() {
 		season++;
 	} else season--;
 
-	if (season >= growSeasonLength + dieSeasonLength || season < 0) seasonUp = !seasonUp;
+	if (season >= growSeasonLength + dieSeasonLength || season < 0) {
+		seasonUp = !seasonUp;
+		if (seasonUp) year++;
+	}
 
-	for (let i in map) {
-		for (let j in map[i]) {
-			if (map[i][j].type == 1) {
+	for (let i in map) {		// This has a big impact on performance. Wouldn't it be possible to do something alike to carykh?
```

Comment:
I think we should only update inactive tiles every 20 frames or so. That'll be easy to implement and it'll cut processing time in this section to 1/20th